Sources/SwiftMetricsBluemix/SwiftMetricsBluemix.swift

```diff
@@ -99,76 +101,58 @@ public class AutoScalar {
   }
 
   private func initCredentials() -> Bool {
-    do {
-      let appEnv = try CloudFoundryEnv.getAppEnv()
-      var autoScalingServiceCreds: [String:Any] = [:]
-      // Find auto-scaling service using convenience method
-      let appEnvServices = appEnv.getServices()
-      for (_, service) in appEnvServices {
-        if service.label.hasPrefix(autoScalingServiceLabelPrefix) {
-          Log.debug("[Auto-Scaling Agent] Found Auto-Scaling service: \(service.name)")
-          autoScalingServiceCreds = service.credentials ?? [:]
-          break
-        }
-      }
-      Log.debug("[Auto-Scaling Agent] Auto-scaling service credentials: \(autoScalingServiceCreds)")
-
-      if autoScalingServiceCreds.isEmpty {
-        Log.error("[Auto-Scaling Agent] Please bind auto-scaling service!")
-        return false
-      }
-
-      // Validate optionals
-      guard let agentUsername = autoScalingServiceCreds["agentUsername"] as? String else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.agentUsername is not found or empty.")
-        return false
-      }
-
-      guard let agentPassword = autoScalingServiceCreds["agentPassword"] as? String else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.agentPassword is not found or empty.")
-        return false
-      }
-
-      guard let appID = autoScalingServiceCreds["app_id"] as? String else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.app_id is not found or empty.")
-        return false
-      }
-
-      guard let host = autoScalingServiceCreds["url"] as? String else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.url is not found or empty.")
-        return false
-      }
-
-      guard let serviceID = autoScalingServiceCreds["service_id"] as? String else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.service_id is not found or empty.")
-        return false
-      }
-
-      // Assign unwrapped values
-      self.host = host
-      self.serviceID = serviceID
-      self.appID = appID
-      self.agentPassword = agentPassword
-      self.agentUsername = agentUsername
-
-      guard let app = appEnv.getApp() else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.url is not found or empty.")
-        return false
-      }
-
-      // Extract fields from App object
-      appName = app.name
-      instanceIndex = app.instanceIndex
-      instanceId = app.instanceId
+    let configMgr = ConfigurationManager().load(.environmentVariables)
+    // Find auto-scaling service using convenience method
+    let scalingServ: Service? = configMgr.getServices(type: autoScalingServiceLabel).first
```

Comment:
@mattcolegate I took a look at the description of this issue: https://github.com/IBM-Swift/SwiftMetrics/issues/33. I agree with what qibobo stated: “the service instance name could be anything but the label name will always be Auto-Scaling”. The label (Auto-Scaling) is set by Bluemix and I don’t believe it can be changed by developers. On the other hand, the name can be anything a developer wants it to be (e.g. my_service, swift, etc.). The following code should then work as expected:

```
 let scalingServ: Service? = configMgr.getServices(type: autoScalingServiceLabel).first
```

It finds the services that have the specified label (or type): Auto-Scaling. It does not use the name.

Having said all this, did I miss something? Let me know if I did.

As for your second question, the method we are using in the PR to get an instance of the AutoScaling service is this constructor: `public init?(withService service: Service)`. This method takes as an argument a Service instance. The method you were probably looking at was this one:  public func getAutoScalingService(name: String) throws -> AutoScalingService. This one requires knowing the name of the service upfront which does not work for what we are trying to do in SwiftMetrics.